### PR TITLE
Rework pack-mode-switcher on windows and add automatability

### LINKED ISSUE #488: [Bug]: pack-mode-switcher doesn't exist in the build
### Issue Summary

I would like to try out Hard mode, but I am not seeing the pack-mode-switcher file in either Curseforge or a GitHub Release file. I [found the commit](https://github.com/ThePansmith/Monifactory/commit/f2e0f44e351f91a6fbc293680a9539b832b70ded) that removed these files from the build due to Curseforge, but could they still exist if you download the pack directly from GitHub?

For now I guess we have to download the appropriate file and put it in our pack manually?

### Reproduction steps

1. Install the pack (I used Prism Launcher)
2. Open the folder of the pack
3. Cannot find the relevant files for switching the mode of the pack.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Mindstyler**
Rewrite pack-mode-switcher.bat as pack-mode-switcher.ps1, increasing code quality and adding the ability to pass command line arguments to switch the modes, making the process automatable on custom launchers.

Add updated instructions how to switch pack modes manually and automatically in README.md.

Add pack-mode-switcher.ps1 back to curseforge distribution. (I couldn't find if Curseforge also disallows .ps1 files, so test it)

Potentially resolves #488 for Windows platform.

### CODE REVIEW

**Comment by Xefyr0** [comment]
Uploading the .bat that does the same thing as the Powershell script [as seen on Discord](https://discord.com/channels/914926812948234260/1229929078547550238/1296494999789375548)
[pack-mode-switcher.txt](https://github.com/user-attachments/files/17436664/pack-mode-switcher.txt)
Mindstyler also made a .batch file that bootstraps the Powershell also in that same discussion.

tl;dr
Now we have 3 options:
1. Direct to Batch (old & ew)
2. Bootstrap in Batch to Powershell (needs 2 files)
3. Powershell directly (requires ExecutionPolicy bypassing)

**Comment by Mindstyler** [comment]
as more info, the 'bootstrap in batch' is simply 1 line of batch code
